internal/proxy/options.go

```diff
@@ -147,8 +147,9 @@ func (o *Options) Validate() error {
 	if o.ClientSecret == "" {
 		msgs = append(msgs, "missing setting: client-secret")
 	}
+	var noDomainOrAddressOverride bool
```

Comment:
This seems like a code smell. Let's specify these as a required default values and allow overrides per upstream.